openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModelTrainer.java

```diff
@@ -343,42 +400,35 @@ static void saveModelFileToDisk(final SWIGTYPE_p_void swigBoosterHandle, final P
     /**
      * Takes the data in dataset and copies it into the features and label C++ arrays through SWIG.
      *
-     * @param dataset              Input train dataset (with target label)
-     * @param swigFeatureDataArray SWIG array of doubles to which the features data will be copied into (in row-major format).
-     * @param swigLabelDataArray   SWIG array of floats to which the labels shall be copied.
+     * @param dataset       Input train dataset (with target label)
+     * @param swigTrainData SWIGTrainData object.
      */
     private static void copyTrainDataToSWIGArrays(final Dataset dataset,
-                                                  final SWIGTYPE_p_double swigFeatureDataArray,
-                                                  final SWIGTYPE_p_float swigLabelDataArray) {
+                                                  final SWIGTrainData swigTrainData) {
 
         final DatasetSchema datasetSchema = dataset.getSchema();
         final int numFields = datasetSchema.getFieldSchemas().size();
-        final int numFeatures = datasetSchema.getPredictiveFields().size();
-        // if target index doesn't exit, return -1.
         final int targetIndex = datasetSchema.getTargetIndex().orElse(-1);
```

Comment:
What do you want to do if the `targetIndex` is not defined? By assigning `-1` you will, most likely, get a `IndexOutOfBoundsException` (at least with the `Instance` implementations that I checked)

Reply:
This cannot happen, the target is ensured in validateForFit, by using the `ValidationUtils`' `validateCategoricalSchema`.

Should I put an if here and raise a RuntimeException too?

Reply:
I would say you can force a `get`, and then add an inline comment with the exact content of your comment